airflow/models/baseoperator.py

```diff
@@ -1575,6 +1577,27 @@ def inherits_from_empty_operator(self):
         # of its subclasses (which don't inherit from anything but BaseOperator).
         return getattr(self, "_is_empty", False)
 
+    def _trigger_timeout(
```

Comment:
do we really need to merge these timeouts in this way?

the execution_timeout is still applied to the trigger, it seems.

so, it seems that sensors could always use the sensor `timeout` value to determine the appropriate deferral timeout to use.  and such could be handled differentially by creating a new trigger subclass.   and let execution_timeout continue to be applied independenty.

it feels a bit wrong to muddle them together.